dev-python/brotlipy/brotlipy-0.7.0.ebuild

```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+
+inherit distutils-r1
+
+DESCRIPTION="Python binding to the Brotli library"
+HOMEPAGE="https://pypi.python.org/pypi/brotlipy"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="${PYTHON_DEPS}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+"
+RDEPEND="${DEPEND}
+	virtual/python-cffi[${PYTHON_USEDEP}]
+"
```

Comment:
1. Run tests.
2. `USE_SHARED_BROTLI=1`, don't use bundled libs.

Reply:
@mgorny After investing a few hours and learning quite a bunch of new things about gentoo and python, I actually do not have much of positive news.

In the tests it uses datasets from brotli (https://github.com/python-hyper/brotlipy/blob/v0.7.0/test/conftest.py#L7 see the TEST_DATA_DIR var). Should I see how I can mock this around and supply somehow the test dataset or should I just give up on the tests? The tests did pass successfully after I manually altered the brotlipy-0.7.0.tar.gz tarball and included necessary files in there. The dataset is a little more than 1Mb of various files that are being compressed/decompressed.

As for USE_SHARED_BROTLI=1 -- unfortunately it was introduced after 0.7.0 tag, i.e. right now it's only in the dev snapshot.

The rest (corrected DEPEND and github homepage) are taken care of by me.

Reply:
Hello! Sorry to have disappeared for some time.

I have invested extra time into this ebuild. As far as the tests - the problem is that brotlipy tests are using data (files to compress/de-compress) from brotli lib. The path it uses (relative to brotlipy workdir is "'libbrotli/tests/testdata". I certainly could patch the source code of brotlipy and change the dirname to point somewhere else. Is there anyway I could "access" contents of "tests/" of a foreign ebuild while testing "my" ebuild? I ran `equery files app-arch/brotli` but it did not show the tests filles anywhere.

I also tried once again to use the shared brotli lib. I patched the latest release of brotlipy with necessary patch to have the option of using external lib instead of the bundled one. The patch worked and I was able to compile brotlipy ebuild. But then a failure expected me :) Brotlipy comes with "0.6.0" brotli bundled whereas "app-arch/brotli" within Gentoo offers only brotli-1.0.* versions. They are incompatible and mitmproxy (a tool that depends on brotlipy) refused to start with errors related to brotli when I tried to launch it.

Should I try to include brotli-0.6.* ebuild into Gentoo repo? Or should we give up and use the bundled one? Also, do you have any suggestion/opinion about the tests? Maybe we should upload the test data files somewhere and download them from SRC_URI?

Reply:
Yes, I understand you.

To use `brotli-1.*` the bottleneck appears to be upstream: brotlipy currently does not support it. Here is an issue on their github where a guy asks if they are planning on doing something about brotli-1.*  https://github.com/python-hyper/brotlipy/issues/129

Does it mean I am out of luck to have dev-python/brotlipy in Gentoo tree? I also double checked the latest mitmproxy release now (remember that I am introducing all these packages for mitmproxy?) and the current master of mitmproxy depends on `brotlipy>=0.7.0,<0.8`. And brotlipy in those versions only works with brotli-0.*

I also got you about the tests. I'll code that part in the coming days :)

Reply:
Michal :) I hope now this ebuild is in proper shape. I just `git push --force` an update into this branch.

A few points about the ebuild:
* I download brotli library in the exact same commit (state) as it's referenced within brotlipy as a git submodule.
* For some reason pypi release of brotlipy did not include `configtest.py` file and it was required for pytest to successfully run the tests, so I placed it into `files` subfolder and copy it into appropriate place during `src_prepare()`.
* I also tried to use github release of brotlipy (which does contain `configtest.py` file) and inject brotli lib into it, but the tests were failing for some reason.

I successfully emerged this ebuild with tests both enabled and disabled.